substrate/frame/staking/src/tests_paged_election.rs

```diff
@@ -0,0 +1,804 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{mock::*, *};
+use frame_support::{assert_ok, testing_prelude::*, BoundedBTreeSet};
+use substrate_test_utils::assert_eq_uvec;
+
+use frame_election_provider_support::{
+	bounds::ElectionBoundsBuilder, ElectionDataProvider, SortedListProvider, Support,
+};
+use sp_staking::StakingInterface;
+
+mod electable_stashes {
+	use super::*;
+
+	#[test]
+	fn add_electable_stashes_work() {
+		ExtBuilder::default().build_and_execute(|| {
+			MaxValidatorSet::set(5);
+			assert_eq!(MaxValidatorSet::get(), 5);
+			assert!(ElectableStashes::<Test>::get().is_empty());
+
+			// adds stashes without duplicates, do not overflow bounds.
+			assert_ok!(Staking::add_electables(vec![1u64, 2, 3].into_iter()));
+			assert_eq!(
+				ElectableStashes::<Test>::get().into_inner().into_iter().collect::<Vec<_>>(),
+				vec![1, 2, 3]
+			);
+
+			// adds with duplicates which are deduplicated implicitly, no not overflow bounds.
+			assert_ok!(Staking::add_electables(vec![1u64, 2, 4].into_iter()));
+			assert_eq!(
+				ElectableStashes::<Test>::get().into_inner().into_iter().collect::<Vec<_>>(),
+				vec![1, 2, 3, 4]
+			);
+
+			// skip final try state checks.
+			SkipTryStateCheck::set(true);
+		})
+	}
+
+	#[test]
+	fn add_electable_stashes_overflow_works() {
+		ExtBuilder::default().build_and_execute(|| {
+			MaxValidatorSet::set(5);
+			assert_eq!(MaxValidatorSet::get(), 5);
+			assert!(ElectableStashes::<Test>::get().is_empty());
+
+			// adds stashes so that bounds are overflown, fails and internal state changes so that
+			// all slots are filled.
+			assert!(Staking::add_electables(vec![1u64, 2, 3, 4, 5, 6].into_iter()).is_err());
```

Comment:
Test the error value here please.

Reply:
done, returning an err with the set of non included stash accounts.